impeller/entity/shaders/gaussian_blur/kernel.glsl

```diff
@@ -16,7 +16,7 @@ struct KernelSample {
 
 uniform KernelSamples {
   int sample_count;
-  KernelSample samples[32];
+  KernelSample samples[48];
```

Comment:
Can we drop this down to 24 by only storing the samples from 0 to ?

I'm slightly concerned that we don't really have any tracking of the uniform size limit across our various backends.

Reply:
For the Pixel 5 the max UBO size is 64KB ( https://vulkan.gpuinfo.org/displayreport.php?id=27816#properties )
The vulkan 1.0 spec requires a minimuimum of 16384B ( https://registry.khronos.org/vulkan/specs/1.0-wsi_extensions/pdf/vkspec.pdf ) 

Assuming 32bit floats:
sizeof(KernelSample) = 12B
sizeof(KernelSample) * 48 = 576 bytes

I don't know if there is a definitive rule about this for opengles, but given the big gap between this an the 1.0 minimum I think we should be good.

Reply:
sizeof(KernelSample) is 16 bytes due to padding requirements. According to https://registry.khronos.org/OpenGL/specs/es/2.0/GLSL_ES_Specification_1.00.pdf

```
 gl_MaxVertexUniformVectors = 128;
 ```
 
 Which assume they mean 4 component float vectors, we're still below the limit. SGTM